Can't use #[faux::methods] on an impl Clone for X {} block

```rust
struct Foo;
#[faux::methods]
impl Clone for Foo {
    fn clone(&self) -> Self {
        Foo
    }
}
```

```
error[E0308]: mismatched types
 --> src/lib.rs:3:1
  |
3 | #[faux::methods]
  | ^^^^^^^^^^^^^^^^ expected struct `_FauxOriginal_Foo`, found struct `Foo`
  |
```
This happens whether or not I manually change `Self -> Foo`. Looking at the expanded source, I think it's because the `methods` macro unconditionally wraps the generated match in `MaybeFaux::Real`, even though it should wrap the result of `call_mock` in `MaybeFaux::Faux`

--- a/faux_macros/src/create.rs
+++ b/faux_macros/src/create.rs
@@ -40,9 +40,7 @@ impl Mockable {
                 }
             };
             let vis = &morphed.vis;
-            syn::Fields::Unnamed(
-                syn::parse2(quote! { (#vis faux::MaybeFaux<#wrapped_self>) }).unwrap(),
-            )
+            syn::Fields::Unnamed(syn::parse_quote! { (#vis faux::MaybeFaux<#wrapped_self>) })
         };
 
         Mockable { real, morphed }

--- a/faux_macros/src/methods.rs
+++ b/faux_macros/src/methods.rs
@@ -108,12 +108,11 @@ impl Mockable {
 
         let (impl_generics, _, where_clause) = generics.split_for_impl();
 
-        let whens = syn::parse2(quote! {
+        let whens = syn::parse_quote! {
             impl #impl_generics #morphed_ty #where_clause {
                 #(#when_methods) *
             }
-        })
-        .unwrap();
+        };
 
         Ok(Mockable {
             real,
@@ -251,7 +250,5 @@ fn publicize_methods(impl_block: &mut syn::ItemImpl) {
             _ => None,
         })
         .filter(|method| method.vis == syn::Visibility::Inherited)
-        .for_each(|mut method| {
-            method.vis = syn::parse2(quote! { pub(super) }).unwrap();
-        });
+        .for_each(|method| method.vis = syn::parse_quote! { pub(super) });
 }

--- a/faux_macros/src/methods/morphed.rs
+++ b/faux_macros/src/methods/morphed.rs
@@ -125,28 +125,28 @@ impl<'a> Signature<'a> {
             Some(path) => quote! { <#real_ty as #path>::#name },
         };
 
-        let block = match &self.method_data {
+        let real_self_arg = if self.method_data.is_some() {
+            // need to pass the real Self arg to the real method
+            Some(syn::Ident::new("r", proc_macro2::Span::call_site()))
+        } else {
+            None
+        };
+        let proxy_args = real_self_arg.iter().chain(arg_idents);
+        let mut proxy_real = quote! { #proxy(#(#proxy_args),*) };
+        if self.is_async {
+            proxy_real.extend(quote! { .await })
+        }
+        if let Some(wrapped_self) = self.wrap_self(morphed_ty, real_self, &proxy_real)? {
+            proxy_real = wrapped_self;
+        }
+
+        let ret = match &self.method_data {
             // not mockable
             // proxy to real associated function
-            None => {
-                let mut proxy_real = quote! { #proxy(#(#arg_idents),*) };
-                if self.is_async {
-                    proxy_real = quote! { #proxy_real.await }
-                }
-
-                self.wrap_self(morphed_ty, real_self, &proxy_real)?
-                    .unwrap_or(proxy_real)
-            }
+            None => syn::parse2(proxy_real).unwrap(),
             // else we can either proxy for real instances
             // or call the mock store for faux instances
             Some(method_data) => {
-                let mut proxy_real = quote! { #proxy(r, #(#arg_idents),*) };
-                if self.is_async {
-                    proxy_real = quote! { #proxy_real.await }
-                }
-                if let Some(wrapped_self) = self.wrap_self(morphed_ty, real_self, &proxy_real)? {
-                    proxy_real = wrapped_self;
-                }
                 let call_mock = if method_data.is_private {
                     quote! { panic!("faux error: private methods are not mockable; and therefore not directly callable in a mock") }
                 } else {
@@ -194,7 +194,10 @@ impl<'a> Signature<'a> {
             }
         };
 
-        Ok(syn::parse2(quote! {{ #block }}).unwrap())
+        Ok(syn::Block {
+            stmts: vec![syn::Stmt::Expr(ret)],
+            brace_token: Default::default(),
+        })
     }
 
     pub fn create_when(&self) -> Option<Vec<syn::ImplItemMethod>> {
@@ -229,12 +232,12 @@ impl<'a> Signature<'a> {
             let last_segment = &output.path.segments.last().unwrap();
             match SelfType::from_path(output) {
                 SelfType::Owned if is_self(output) => Some(match real_self {
-                    SelfType::Owned => quote! {{ Self(faux::MaybeFaux::Real(#block))}},
+                    SelfType::Owned => quote! { Self(faux::MaybeFaux::Real(#block)) },
                     generic => {
                         let new_path = generic
                             .new_path()
                             .expect("Generic self should have new() function");
-                        quote! {{ Self(faux::MaybeFaux::Real(#new_path(#block)))}}
+                        quote! { Self(faux::MaybeFaux::Real(#new_path(#block))) }
                     }
                 }),
                 generic if self_generic(&last_segment.arguments) => {
@@ -242,7 +245,7 @@ impl<'a> Signature<'a> {
                         let new_path = real_self
                             .new_path()
                             .expect("return type should not be Self");
-                        Some(quote! {{ #new_path(Self(faux::MaybeFaux::Real(#block)))}})
+                        Some(quote! { #new_path(Self(faux::MaybeFaux::Real(#block))) })
                     } else {
                         return Err(darling::Error::custom(wrong_self_type_error(
                             generic, real_self,

--- a/faux_macros/src/methods/morphed.rs
+++ b/faux_macros/src/methods/morphed.rs
@@ -279,10 +279,10 @@ impl<'a> MethodData<'a> {
         let faux_ident =
             syn::Ident::new(&format!("_faux_{}", name), proc_macro2::Span::call_site());
 
-        let empty = syn::parse2(quote! { () }).unwrap();
+        let empty = syn::parse_quote! { () };
         let output = output.unwrap_or(&empty);
 
-        let when_method = syn::parse2(quote! {
+        let when_method = syn::parse_quote! {
             pub fn #when_ident(&mut self) -> faux::When<#receiver_tokens, (#(#arg_types),*), #output, faux::when::Any> {
                 match &mut self.0 {
                     faux::MaybeFaux::Faux(faux) => faux::When::new(
@@ -292,16 +292,14 @@ impl<'a> MethodData<'a> {
                     faux::MaybeFaux::Real(_) => panic!("not allowed to mock a real instance!"),
                 }
             }
-        })
-        .unwrap();
+        };
 
-        let faux_method = syn::parse2(quote! {
+        let faux_method = syn::parse_quote! {
             #[allow(clippy::needless_arbitrary_self_type)]
             pub fn #faux_ident(self: #receiver_tokens, input: (#(#arg_types),*)) -> #output {
                 panic!("do not call this")
             }
-        })
-        .unwrap();
+        };
 
         vec![when_method, faux_method]
     }

--- a/faux_macros/src/methods/receiver.rs
+++ b/faux_macros/src/methods/receiver.rs
@@ -47,7 +47,7 @@ impl Receiver {
         self_type: SelfType,
         proxy_real: TokenStream,
         call_mock: TokenStream,
-    ) -> darling::Result<TokenStream> {
+    ) -> darling::Result<syn::Expr> {
         let get_self = match &self.kind {
             SelfKind::Owned
             | SelfKind::Pointer(PointerKind::Ref)
@@ -170,7 +170,7 @@ impl Receiver {
             }
         };
 
-        Ok(quote! {
+        Ok(syn::parse_quote! {
             match #get_self {
                 Self(faux::MaybeFaux::Real(r)) => { #proxy_real },
                 Self(faux::MaybeFaux::Faux(q)) => { #call_mock },

